src/transformers/models/oneformer/processing_oneformer.py

```diff
@@ -48,22 +69,23 @@ class OneFormerProcessor(ProcessorMixin):
     tokenizer_class = ("CLIPTokenizer", "CLIPTokenizerFast")
 
     def __init__(
-        self, image_processor=None, tokenizer=None, max_seq_length: int = 77, task_seq_length: int = 77, **kwargs
+        self,
+        image_processor=None,
+        tokenizer=None,
+        max_seq_length: int = 77,
+        task_seq_length: int = 77,
     ):
         if image_processor is None:
             raise ValueError("You need to specify an `image_processor`.")
         if tokenizer is None:
             raise ValueError("You need to specify a `tokenizer`.")
 
-        self.max_seq_length = max_seq_length
-        self.task_seq_length = task_seq_length
-
         super().__init__(image_processor, tokenizer)
 
-    def _preprocess_text(self, text_list=None, max_length=77):
-        if text_list is None:
-            raise ValueError("tokens cannot be None.")
+        self.max_seq_length = max_seq_length
+        self.task_seq_length = task_seq_length
 
+    def _preprocess_text(self, text_list: PreTokenizedInput, max_length: int = 77):
```

Comment:
so this tokenizes which should be done with `self.tokenizer(..., output_kwargs['text_kwargs'])`, and it seems that it puts pad tokens where the attention mask is 0 - this should be covered by the tokenizer already, would be a nice refactor

Reply:
I will look into this...

Reply:
Atm I don't see how this can be simplified because the function is called with different arguments inside `__call__`, and it is also called in `encode_inputs`. Any suggestions?

Reply:
Agreed that it would be best to use self.tokenizer directly, and pass in the other text kwargs as well. For max_length, you can do the following:
- if the max_length kwarg is not set, set it to `max_seq_length` or `task_seq_length` in the two different calls.
- if it is set, use its value for both task and seq.

Reply:
I tried this (commit 3c7724d), but I don't see yet how it is simpler/more readable :)
Couple of observations:

- the function `_preprocess_text` is called both with `task_seq_length` and `max_seq_length`, so if I add `"max_length"` to the `text_kwargs` dict, I would need to add and then replace in `__call__`.
- in order to do something similar to what the attention-mask is doing (padding with 0s), I would need to change `self.tokenizer.pad_token_id` (and then change it back to what was before, e.g. 49407 (endoftext), because I can't pass in `pad_token_id` to the tokenizer. I am most probably missing here something, so suggestions welcome.